src/fides/api/util/dict_diff.py

```diff
@@ -0,0 +1,46 @@
+from typing import Any, Dict, Tuple
+
+
+def dict_diff(
+    first_dict: Dict[str, Any], second_dict: Dict[str, Any]
+) -> Tuple[Dict[str, Any], Dict[str, Any]]:
```

Comment:
Did you look at any existing libraries? There's a built-in deepdiff, just the first one I saw.  There's some tradeoffs, the output here is pretty verbose but on the flipside, I think the handling of nested data and null values in our custom `dict_diff` could be improved too.  There may be an existing library that already does exactly what we need.

```python
>>> import deepdiff
>>> dict_1 = {"a": {"b": "c"}}
>>> dict_2 = {"a": {"b": "d"}}
>>> deepdiff.diff.DeepDiff(dict_1, dict_2)
{'values_changed': {"root['a']['b']": {'new_value': 'd', 'old_value': 'c'}}}
>>> deepdiff.diff.DeepDiff(dict_1, dict_1)
{}

>>> dict_3 = {"a": {"b": "d", "c": [1, 2]}}
>>> deepdiff.diff.DeepDiff(dict_2, dict_3)
{'dictionary_item_added': [root['a']['c']]}

>>> dict_4 =  {"a": {"b": "d", "c": None}}
>>> deepdiff.diff.DeepDiff(dict_3, dict_4)
{'type_changes': {"root['a']['c']": {'old_type': <class 'list'>, 'new_type': <class 'NoneType'>, 'old_value': [1, 2], 'new_value': None}}}

```